editor/plugins/tile_map_editor_plugin.cpp

```diff
@@ -339,9 +339,9 @@ void TileMapEditor::_set_cell(const Point2i &p_pos, Vector<int> p_values, bool p
 	if (manual_autotile || (p_value != -1 && node->get_tileset()->tile_get_tile_mode(p_value) == TileSet::ATLAS_TILE)) {
 		if (current != -1) {
 			node->set_cell_autotile_coord(p_pos.x, p_pos.y, position);
-		} else if (node->get_tileset()->tile_get_tile_mode(p_value) == TileSet::ATLAS_TILE && priority_atlastile) {
 			// BIND_CENTER is used to indicate that bitmask should not update for this tile cell.
 			node->get_tileset()->autotile_set_bitmask(p_value, Vector2(p_pos.x, p_pos.y), TileSet::BIND_CENTER);
+		} else if (node->get_tileset()->tile_get_tile_mode(p_value) == TileSet::ATLAS_TILE && priority_atlastile) {
```

Comment:
This change causes a regression in the editor. When you place some tiles with priority disabled, after changing to enable it stops working.
![out](https://user-images.githubusercontent.com/1387165/82121829-55ca4880-9766-11ea-8846-54e6c0673171.gif)
This part of the code is very sensitive and should be changed very carefully.

> Reason: ATLAS Tileset by itself does not allow to set manual bitmasks via editor (which makes sense), but the bitmask code requires a "center bitmask flag" (as described above).

Alternatively, you can set bitmaks by code like this:
```gdscript
extends TileMap

func _ready():
	randomize()

	for x in range(100):
		for y in range(100):
			set_cell(x, y, 0)
			tile_set.autotile_set_bitmask(0, Vector2(x, y), TileSet.BIND_CENTER)

	update_bitmask_region()
```
This may not be very obvious and is not documented anywhere but will do the trick.

Reply:
> and is not documented anywhere

If the PR ends up not being merged, this should be documented in either the `TileMap` or `TileSet` docs.

Reply:
The only problem is that this is a hack. I'm not sure if documenting a hack is a good idea. If I can think of another way to fix this, I try to remove this hack.

Reply:
@guilhermefelipecgs But wasn't this also the case before my change?
And if I got it right, you already provided a solution for this with the new PR #38790 

Or am I wrong?

Reply:
Internally the editor sets `BIND_CENTER` for `ATLAS_TILE`, this was already a hack.

My PR only fix the editor, when using eraser tool. If you want to use `priority` with atlas in gdscript, you still need to call this `tile_set.autotile_set_bitmask(0, Vector2(x, y), TileSet.BIND_CENTER)`. But this call shouldn't be necessary, an proper fix still needs to be made.

Reply:
@guilhermefelipecgs Yes, but above you exactly mention the same bug as you fixed as a regression issue. And here it's a show stopper for this PR, am I wrong?

I will test your fix later, but I'm pretty sure that this will also work in the inverted solution (my PR)

Reply:
My PR didn't introduce any regression in my tests, the gif I posted here https://github.com/godotengine/godot/pull/38546#discussion_r426158190 was tested with your patch. The gdscript solution is a hack to get it working, the same hack is used in the godot code. I am refactoring this code (I'll open a new PR for this) to completely remove the hack and make it work for both the editor and gdscript.

Reply:
@zzz-assault PR #38801 open, in case you want to test it.

Reply:
Great … but if you anyhow completley change the code, I'll close this PR.

Doesn't make sense to have to have a PR for solving an old code base.

Big THANKS for your efforts :)